packages/utils/src/cmds/schema/diff/index.js

```diff
@@ -0,0 +1,269 @@
+    const schemasUrl = url.concat(defaultOpts.schemasEndpoint)
+    const infoUrl = url.concat(defaultOpts.infoEndpoint)
+    const requestOpts = { ...defaultRequestOpts, baseUrl }
+    requestOpts.headers.Authorization = await getAuthHeader(url, restArgs)
+
+    const meta = await asyncRequest(infoUrl, requestOpts)
+    const schemaFileName = `${schemaIdentifier(meta)}.json`
+    const loc = await cache.get(schemasUrl, schemaFileName, {
+        raw: true,
+        requestOpts,
+        force: restArgs.force,
+    })
+    const schemas = getJSONFile(loc)
+
+    return {
+        meta,
+        schemas,
+    }
+}
+
+async function getSchemas(urlLike, baseUrl, restArgs) {
+    let schemas
+    let fileContents
+
+    if ((fileContents = getJSONFile(urlLike))) {
+        reporter.debug(urlLike, ' is a file')
+        schemas = fileContents
+    } else {
+        schemas = await schemasFromUrl(urlLike, baseUrl, restArgs)
+    }
+
+    if (!schemas.schemas || !schemas.meta) {
+        reporter.error(
+            `${urlLike} is malformed. Must have an object with keys: [schemas, meta]`
+        )
+        process.exit(1)
+    }
+    return schemas
+}
+
+function diff(left, right, html = false, json) {
+    const delta = Differ.diff(left.schemas, right.schemas)
+
+    if (json) {
+        reporter.pipe(JSON.stringify(delta))
+    } else if (html !== false) {
+        writeHtml(html, left, right, delta)
+    }
+
+    return delta
+}
+
+function generateHtml(left, delta, meta) {
+    const assets = {
+        jsondiffpatchCSS: utils.btoa(
+            fs.readFileSync(
+                require.resolve('jsondiffpatch/dist/formatters-styles/html.css')
+            )
+        ),
+        jsondiffpatchJS: utils.btoa(
+            fs.readFileSync(
+                require.resolve('jsondiffpatch/dist/jsondiffpatch.umd.slim.js')
+            )
+        ),
+    }
+
+    const template = fs
+        .readFileSync(path.join(__dirname, 'index.ejs'))
+        .toString()
+
+    return ejs.render(template, {
+        left,
+        delta,
+        meta,
+        ...assets,
+    })
+}
+
+function writeHtml(loc, left, right, delta) {
+    reporter.debug('Generating visuals...')
+
+    const html = generateHtml(left.schemas, delta, {
+        left: left.meta,
+        right: right.meta,
+    })
+    const isDir = typeof loc === 'string' ? fs.statSync(loc).isDirectory : false
+    if (loc === '' || loc === true || isDir) {
+        const fileName = `${schemaDiffIdentifier(left.meta, right.meta)}.html`
+        loc = path.join(isDir ? loc : '', fileName)
+    }
+
+    fs.writeFileSync(loc, html)
+    reporter.info('Visual output generated:', loc)
+    reporter.pipe(loc)
+}
+
+async function run({ url1, url2, baseUrl, html, json, ...rest }) {
+    cache = rest.getCache()
+    const left = await getSchemas(url1, baseUrl, rest)
+    const right = await getSchemas(url2, baseUrl, rest)
+    //let [left, right] = await Promise.all([prom1, prom2])
```

Comment:
I wanted to do this to send them concurrently. However, in the case of --prompt-auth we need to wait for the prompt for each request. Opted for simplicity instead of extra logic to fix this.

Reply:
Seems OK for now, might be nice to parallelize in the future